reporter/otlp_reporter.go

```diff
@@ -255,24 +184,24 @@ func (r *OTLPReporter) ReportCountForTrace(_ libpf.TraceHash, _ uint16, _ *Trace
 // ExecutableKnown returns true if the metadata of the Executable specified by fileID is
 // cached in the reporter.
 func (r *OTLPReporter) ExecutableKnown(fileID libpf.FileID) bool {
-	_, known := r.executables.Get(fileID)
+	_, known := r.pdata.Executables.Get(fileID)
 	return known
 }
 
 // ExecutableMetadata accepts a fileID with the corresponding filename
 // and caches this information.
 func (r *OTLPReporter) ExecutableMetadata(args *ExecutableMetadataArgs) {
-	r.executables.Add(args.FileID, execInfo{
-		fileName:   args.FileName,
-		gnuBuildID: args.GnuBuildID,
+	r.pdata.Executables.Add(args.FileID, samples.ExecInfo{
+		FileName:   args.FileName,
+		GnuBuildID: args.GnuBuildID,
 	})
 }
 
 // FrameKnown returns true if the metadata of the Frame specified by frameID is
 // cached in the reporter.
 func (r *OTLPReporter) FrameKnown(frameID libpf.FrameID) bool {
 	known := false
-	if frameMapLock, exists := r.frames.Get(frameID.FileID()); exists {
+	if frameMapLock, exists := r.pdata.Frames.Get(frameID.FileID()); exists {
```

Comment:
Fix from https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/256:
```suggestion
	if frameMapLock, exists := r.pdata.Frames.GetAndRefresh(frameID.FileID(), ...); exists {
```